src/psyclone/tests/test_files/dynamo0p3/ru_kernel_mod.f90

```diff
@@ -1,29 +1,102 @@
+    integer :: iterates_over = CELLS
+    integer :: gh_shape = gh_quadrature_XYoZ
+  contains
+    procedure, nopass :: ru_code
+  end type
 
-type, public, extends(kernel_type) :: ru_kernel_type
-  private
-  type(arg_type) :: meta_args(6) = (/                                  &
-       arg_type(GH_FIELD,   GH_INC,  W2),                              &
-       arg_type(GH_FIELD,   GH_READ, W3),                              &
-       arg_type(GH_INTEGER, GH_READ),                                  &
-       arg_type(GH_REAL,    GH_READ),                                  &
-       arg_type(GH_FIELD,   GH_READ, W0),                              &
-       arg_type(GH_FIELD*3, GH_READ, W0)                               &
-       /)
-  type(func_type) :: meta_funcs(3) = (/                                &
-       func_type(W2, GH_BASIS, GH_DIFF_BASIS),                         &
-       func_type(W3, GH_BASIS),                                        &
-       func_type(W0, GH_BASIS, GH_DIFF_BASIS)                          &
-       /)
-  integer :: iterates_over = CELLS
-  integer, parameter :: gh_shape = gh_quadrature_XYoZ
-contains
-  procedure, nopass ::ru_code
-end type
+  contains
 
-contains
-
-subroutine ru_code()
+  subroutine ru_code(nlayers, field_1_w2, field_2_w3, iscalar_3, rscalar_4, field_5_w0, &
+       field_6_w0_v1, field_6_w0_v2, field_6_w0_v3, ndf_w2, undf_w2, map_w2, basis_w2,  &
+       diff_basis_w2, ndf_w3, undf_w3, map_w3, basis_w3, ndf_w0, undf_w0, map_w0,       &
+       basis_w0, diff_basis_w0, np_xy, np_z, weights_xy, weights_z)
+      USE constants_mod, ONLY: r_def
+      IMPLICIT NONE
+      INTEGER, intent(in) :: nlayers
+      INTEGER, intent(in) :: ndf_w2
+      INTEGER, intent(in) :: undf_w2
+      INTEGER, intent(in) :: ndf_w3
+      INTEGER, intent(in) :: undf_w3
+      INTEGER, intent(in) :: ndf_w0
+      INTEGER, intent(in) :: undf_w0
+      REAL(KIND=r_def), intent(inout), dimension(undf_w2) :: field_1_w2
+      REAL(KIND=r_def), intent(in), dimension(undf_w3) :: field_2_w3
+      INTEGER, intent(in) :: iscalar_3
+      REAL(KIND=r_def), intent(in) :: rscalar_4
+      REAL(KIND=r_def), intent(in), dimension(undf_w0) :: field_5_w0
+      REAL(KIND=r_def), intent(in), dimension(undf_w0) :: field_6_w0_v1
+      REAL(KIND=r_def), intent(in), dimension(undf_w0) :: field_6_w0_v2
+      REAL(KIND=r_def), intent(in), dimension(undf_w0) :: field_6_w0_v3
+      INTEGER, intent(in), dimension(ndf_w2) :: map_w2
+      REAL(KIND=r_def), intent(in), dimension(3,ndf_w2,np_xy,np_z) :: basis_w2
+      REAL(KIND=r_def), intent(in), dimension(1,ndf_w2,np_xy,np_z) :: diff_basis_w2
+      INTEGER, intent(in), dimension(ndf_w3) :: map_w3
+      REAL(KIND=r_def), intent(in), dimension(1,ndf_w3,np_xy,np_z) :: basis_w3
+      INTEGER, intent(in), dimension(ndf_w0) :: map_w0
+      REAL(KIND=r_def), intent(in), dimension(1,ndf_w0,np_xy,np_z) :: basis_w0
+      REAL(KIND=r_def), intent(in), dimension(3,ndf_w0,np_xy,np_z) :: diff_basis_w0
```

Comment:
Changes to make kernel compilable? Thanks.

Reply:
Yes that's right.